### PR TITLE
Firebase animated list index

### PR DESCRIPTION
**Pull request by pcomans**
See: https://github.com/flutter/flutter/issues/11760

### CODE REVIEW

**Comment by pcomans** [comment]
I signed it!

**Comment by pcomans** [comment]
I signed it!

**Comment by sethladd** [comment]
cc @kroikie @collinjackson to review.

**Comment by kroikie** [comment]
Looks like a good breaking change to me, @pcomans please add a note to the CHANGELOG and bump the version in the pubspec file.

File: packages/firebase_database/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.0.15
```
Comment (goderbauer) [inline packages/firebase_database/CHANGELOG.md]: For a breaking change you should bump the version to 0.1.0. That way, people who have pinned their usage of this plugin to "^0.0.14" will not be auto-upgraded into a broken state.

Comment (pcomans) [reply packages/firebase_database/CHANGELOG.md]: @goderbauer Thank you!

**Comment by pcomans** [comment]
@goderbauer I bumped the version to 0.1.0.

**Comment by pcomans** [comment]
@goderbauer @sethladd are we OK to merge this? :-)

**Comment by pcomans** [comment]
@collinjackson thanks for merging! We are using FirebaseAnimatedList quite extensively and would love to contribute / give feedback.

**Comment by lukepighetti** [comment]
@pcomans Hello, I'm trying to access the next item in the list, is there a way to do that currently? If not, any thoughts before I make a PR for it?

**Comment by pcomans** [comment]
Hi Luke, in the past my team solved similar issues by fetching the list
from Firebase separately and pre-processing the data. We iterated over the
data and added information about the next / previous element.
I'm curious about your solution. Feel free to tag me on the PR if you open
it.

**Comment by lukepighetti** [comment]
@pcomans Thanks for getting back to me, you have inspired me to create a new PR. (My last one was not tested so I removed it.) #730 I am using this to create stacked chats and it works well.

Note: I was able to get this functionality by saving previousSnap and looking back one and saving the new one as previousSnap... without forking firebase_animated_list.dart.... but what I found is that this was unable to handle hot reloads so I went back to my custom version.

**Pull request merged**